arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -196,4 +320,77 @@ modifier onlyFromMyL1Contract() override {
+| **Where is the contract's state stored?**                                  | • On L1 nodes<br />• Each Ethereum node stores the full state                                                                                       | • On L2 nodes<br />• Each Arbitrum node stores the full state                                                                                                 |
+| **How does the contract deployment impact L1's physical state?**           | • Increases storage size<br />• Changes account balances as per the contract deployment cost                                                        | • Indirectly, by generating a commitment on L1<br />• The contract's bytecode, deployed on L2, results in a Merkle proof on L1                                |
+| **How does the contract deployment impact L2's physical state?**           | • Indirectly, if the L1 contract interacts with L2<br />• Changes in L1 contracts could be relayed to L2 via Retryable Tickets                      | • Increases storage size<br />• Changes account balances as per the contract deployment cost                                                                  |
+| **When is gas paid?**                                                      | • During contract deployment<br />• During contract execution                                                                                       | • During contract deployment<br />• During contract execution                                                                                                 |
+| **Where is gas paid?**                                                     | • On Ethereum L1<br />• Paid to Ethereum validators                                                                                                 | • On Arbitrum L2<br />• Paid to Arbitrum validators                                                                                                           |
+| **Who pays gas?**                                                          | • Contract deployer<br />• Transaction initiator                                                                                                    | • Contract deployer<br />• Transaction initiator                                                                                                              |
+| **Can another L1 contract call the deployed contract's methods directly?** | • Yes<br />• Direct interaction between L1 contracts is possible                                                                                    | • No<br />• L1 contracts can't directly call methods of L2 contracts<br />• But can interact via Gateway contracts and Retryable Tickets                      |
```

Comment:
> But can interact via Gateway contracts and Retryable Tickets

I would avoid talking about Gateway contracts on this page, since that's specific to the token bridge (which is a particular dapp that happens to use retryable tickets / L2-to-L1 messages, but i think we want to keep this to x-chain messaging more generally)